kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -36,54 +36,22 @@ sealed trait IOs extends Effect[IOs]:
         }
 
     def attempt[T, S](v: => T < S): Try[T] < S =
-        def attemptLoop(v: T < S): Try[T] < S =
-            v match
-                case kyo: Suspend[MX, Any, T, S] @unchecked =>
-                    new Continue[MX, Any, Try[T], S](kyo):
-                        def apply(v: Any < S, s: Safepoint[S], l: Locals.State) =
-                            try
-                                attemptLoop(kyo(v, s, l))
-                            catch
-                                case ex: Throwable if (NonFatal(ex)) =>
-                                    Failure(ex)
-                case _ =>
-                    Success(v.asInstanceOf[T])
-        try
-            val v0 = v
-            if isNull(v0) then
-                throw new NullPointerException
-            attemptLoop(v0)
-        catch
-            case ex: Throwable if (NonFatal(ex)) =>
-                Failure(ex)
-        end try
-    end attempt
+        eval(v.map(Success(_): Try[T]))((s, k) =>
+            try k()
+            catch
+                case ex if NonFatal(ex) =>
+                    Failure(ex)
+        )
```

Comment:
This method only specifies `resume`, both `done` and `suspend` use no-op default parameters.